### DISCUSSION TITLE
Run multiple queries with a single HTTP request

### DISCUSSION THREAD

**Discussion by Nick-Mazuk** [supabase/supabase#900] [Questions] [answered]
Is it currently possible to run multiple SQL queries with a single HTTP request with Supabase?

For instance, if you have a dashboard overview page, it's totally possible to need to query multiple tables independently to provide a wholistic overview of someone's data. And with queries that, to my knowledge, aren't idiomatic to SQL's table joins.

For instance, the best way I know to query multiple tables simultaneously is:

```ts
const [response1, response2, response3] = await Promise.all([
    supabase.from('table1').select('*'),
    supabase.from('table2').select('*'),
    supabase.from('table3').select('*'),
])
```

But, I believe that results in 3 HTTP requests.

**Comment by steve-chavez** [comment, marked as answer]
> it's totally possible to need to query multiple tables independently to provide a wholistic overview of someone's data. And with queries that, to my knowledge, aren't idiomatic to SQL's table joins.

I see two options here:

- Create a VIEW [that has UNIONs](https://www.postgresqltutorial.com/postgresql-union/) of the tables you need. Then you can request that VIEW like any other table with [supabase.from('view')](https://supabase.io/docs/client/select).
- Create a FUNCTION that returns a dynamic table. And then you call that function through [supabase.rpc('all_data')](https://supabase.io/docs/client/rpc).

```sql
-- sample dynamic table function
create function all_data(user_id uuid)
returns table(table1_col int, table2_col text, table3_col text) -- can have different types of columns coming from different tables
as $$
-- here you can use CTEs, FOR LOOPS, variables, etc
return ...;
$$ language plpgsql;
```

Both would be done in one http request.
